code/modules/clothing/head/helmet.dm

```diff
@@ -115,3 +139,147 @@ obj/item/clothing/head/helmet/bluetaghelm
 	armor = list(melee = 30, bullet = 10, laser = 20,energy = 10, bomb = 20, bio = 0, rad = 0)
 	// Offer about the same protection as a hardhat.
 	flags_inv = HIDEEARS|HIDEEYES
+	canAttachCam =  0
+
+//LightToggle
+
+/obj/item/clothing/head/helmet/update_icon()
+
+	var/state = "[initial(icon_state)]"
+	if(helmetCam)
+		state += "-cam" //"helmet-cam"
+	if(F)
+		if(F.on)
+			state += "-flight-on" //"helmet-flight-on" // "helmet-cam-flight-on"
+		else
+			state += "-flight" //etc.
+
+	icon_state = state
+
+	usr.update_inv_head(0)
+	return
+
+/obj/item/clothing/head/helmet/ui_action_click()
+	toggle_helmlight()
+
+/obj/item/clothing/head/helmet/attackby(var/obj/item/A as obj, mob/user as mob, params)
+	if(istype(A, /obj/item/device/flashlight/seclite))
+		var/obj/item/device/flashlight/seclite/S = A
+		if(can_flashlight)
+			if(!F)
+				user.drop_item()
+				user << "<span class='notice'>You click [S] into place on [src].</span>"
+				if(S.on)
+					SetLuminosity(0)
+				F = S
+				A.loc = src
+				update_icon()
+				update_helmlight(user)
+				verbs += /obj/item/clothing/head/helmet/proc/toggle_helmlight
+		return
+
+	if(istype(A, /obj/item/weapon/screwdriver))
+		if(F)
+			for(var/obj/item/device/flashlight/seclite/S in src)
+				user << "<span class='notice'>You unscrew the seclite from [src].</span>"
+				F = null
+				S.loc = get_turf(user)
+				update_helmlight(user)
+				S.update_brightness(user)
+				update_icon()
+				usr.update_inv_head(0)
+				verbs -= /obj/item/clothing/head/helmet/proc/toggle_helmlight
+			return
+
+
+	if(istype(A, /obj/item/weapon/camera_assembly))
+		if(!canAttachCam)
+			user << "<span class='warning'>You can't attach [A] to [src]!</span>"
+			return
+		if(helmetCam)
+			user << "<span class='notice'>[src] already has a mounted camera.</span>"
+			return
+		user.drop_item()
+		helmetCam = new /obj/machinery/camera(src)

```

Comment:
As someone who has worked on the AI freeview, and basically got it working, functional and ported long ago, I can tell you that putting a camera in a object that is going to move is going to cause a bug where the camera is only going to appear on the AI view for where it was built.

Robots have mobile cameras too but they have a mechanic where, everytime they move, they trigger an event to update the AI's camera vision.

Possible solutions:

* Disable the AI freeview from seeing the security helmet vision, thus making it security camera computer only.
* Add in code to update the camera's position, in the AI's freeview, whenever the player moves. WARNING: This can be a performance hog.